ui/app/AppLayouts/Chat/controls/community/CommunityPermissionsRow.qml

```diff
@@ -0,0 +1,315 @@
+                   (holdingsCount1 === 1 && holdingsCount2 === 2)) {
+                    shortModel.append(permission2)
+                }
+            }
+        }
+    }
+
+    implicitHeight: 24
+    spacing: 4
+    padding: 1
+
+    background: Rectangle {
+        color: root.backgroundColor
+        radius: root.backgroundRadius
+    }
+
+    contentItem: RowLayout {
+        id: container
+
+        anchors.centerIn: parent
+        anchors.margins: root.padding
+        spacing: root.spacing
+
+        StatusIcon {
+            Layout.preferredHeight: container.height - 6
+            Layout.preferredWidth: Layout.preferredHeight
+            Layout.leftMargin: 4
+
+            icon: root.requirementsMet ? "tiny/unlocked" : "tiny/locked"
+            color: Theme.palette.baseColor1
+        }
+
+        Repeater {
+            id: repeater
+
+            model: shortModel
+
+            RowLayout {
+                spacing: root.spacing
+
+                SinglePermissionRow {
+                    model: holdingsListModel
+                }
+
+                StatusBaseText {
+                    visible: index !== (repeater.count - 1) || d.dotsVisible
+                    text: qsTr("or")
+                    font.pixelSize: root.pixelSize
+                    color: Theme.palette.baseColor1
+                }
+            }
+        }
+
+        StatusRoundedComponent {
+            Layout.preferredHeight: container.height
+            Layout.preferredWidth: Layout.preferredHeight
+
+            visible: d.dotsVisible
+            color: Theme.palette.baseColor3
+            border.color: root.backgroundColor
+            border.width: root.overlappingBorder
+
+            StatusIcon {
+                anchors.centerIn: parent
+                visible: d.dotsVisible
+                icon: "dots-icon"
+                height: root.dotsIconSize
+                width: height
+            }
+        }
+    }
+
+    onModelChanged: d.buildShortModel(root.model)
+    Component.onCompleted: d.buildShortModel(root.model)
+
+    ListModel{ id: shortModel }
+
+    component SinglePermissionRow :  RowLayout {
+        id: singlePermissionItem
+
+        readonly property int maxVisualTokens: 3
+
+        property var model
+        property string plusElementText: ""
+        property bool plusElementVisible: false
+
+        function buildTokensRowModel(model) {
+            shortTokensRowModel.clear()
+            if(!model)
+                return
+
+            var modelCount = model.rowCount()
+
+            // Need of shorter model
+            if(modelCount > maxVisualTokens) {
+                for(var i = 0; i < singlePermissionItem.maxVisualTokens - 1; i++)
+                    shortTokensRowModel.append(ModelUtils.get(model, i))
+
+                // Set Plus element properties:
+                plusElementVisible = true
+                if(modelCount === 4)
+                    plusElementText = qsTr("+2")
```

Comment:
+1 for @alexjba comment about that. Please note that limit of 5 can be changed in the future and then this code won't work properly. I also wonder if that string requires i18n. It seems to have universal symbolic meaning and therefore doesn't need translations.